### PR TITLE
Stats: Adjust post detail layout for FixedNavigationHeader and Highlight cards

### PR DESCRIPTION
**Pull request by dognose24**
#### Proposed Changes

* Adjust `FixedNavigationHeader` to fit the modernized layout (`max-width: 1224px`).
* Add `min-width` to Highlight cards for scrolling on the narrower viewport.
* Modify Highlight card block styles on mobile.

#### Testing Instructions

* Spin up this change with the Calypso Live link.
* Navigate to the Stats `Post Detail` page (`/stats/post/${post-id}/${site-id}`).
* Ensure the post detail page works as previously.
* Append the feature flag `?flags=stats/enhance-post-detail` to the URL.
* Ensure the layout of the fixed navigation header aligns with other Stats pages.

| Before | After |
| --- | --- |
| 截圖 2022-12-27 下午10 32 37 | 截圖 2022-12-27 下午10 32 57 |

* Ensure the Highlight cards look reasonable on narrower and mobile viewports.

| Narrower | Mobile |
| --- | --- |
| 截圖 2022-12-27 下午10 49 20 | 截圖 2022-12-27 下午10 48 37 |

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70671

### CODE REVIEW

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Unfortunately, this change appears to break the post details page when the feature flag is disabled:

|Production|Without Feature Flag (Broken)|With Feature Flag|
|-|-|-|
|image|image|image|

Note these visual regressions:
- Chart's y-axis labels flowing outside of the container
image

- Tables taking up the entire width `#primary.layout__primary` instead of being limited to a maximum width of 1224px.
image

**Comment by dognose24** [comment]
> Unfortunately, this change appears to break the post details page when the feature flag is disabled:

> Note these visual regressions:

All fixed in [7127bb0](https://github.com/Automattic/wp-calypso/pull/71530/commits/7127bb0e118db2166bc709d56edbc80a3bf6357d).

**Review by keoshi — APPROVED** [review]
LGTM!

File: client/my-sites/stats/stats-post-detail/index.jsx [hunk 1]
```diff
@@ -139,7 +139,90 @@ class StatsPostDetail extends Component {
...
+						{ showViewLink && (
+							<Button onClick={ this.openPreview }>
+								<span>{ actionLabel }</span>
+							</Button>
+						) }
+					</FixedNavigationHeader>
+
+					{ isFeatured && (
+						<PostDetailHighlightsSection siteId={ siteId } postId={ postId } post={ post } />
+					) }
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-post-detail/index.jsx]:
```suggestion
					<PostDetailHighlightsSection siteId={ siteId } postId={ postId } post={ post } />
```

File: client/my-sites/stats/stats-post-detail/index.jsx [hunk 2]
```diff
@@ -139,7 +139,90 @@ class StatsPostDetail extends Component {
...
+						/>
+					) }
+
+					{ ! isLoading && countViews > 0 && (
+						<>
+							<PostSummary siteId={ siteId } postId={ postId } />
+
+							{ ! isFeatured && !! postId && (
+								<PostLikes siteId={ siteId } postId={ postId } postType={ postType } />
+							) }
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-post-detail/index.jsx]:
```suggestion
```

File: client/my-sites/stats/stats-post-detail/index.jsx [hunk 3]
```diff
@@ -139,7 +139,90 @@ class StatsPostDetail extends Component {
...
+
+							{ isFeatured && <PostDetailTableSection siteId={ siteId } postId={ postId } /> }
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-post-detail/index.jsx]:
```suggestion
							<PostDetailTableSection siteId={ siteId } postId={ postId } />
```

File: client/my-sites/stats/stats-post-detail/index.jsx [hunk 4]
```diff
@@ -139,7 +139,90 @@ class StatsPostDetail extends Component {
...
+										dataKey="averages"
+										title={ translate( 'Average per day' ) }
+										total={ translate( 'Overall' ) }
+										siteId={ siteId }
+										postId={ postId }
+									/>
+
+									<PostWeeks siteId={ siteId } postId={ postId } />
+								</>
+							) }
```
Comment (jsnmoon) [inline client/my-sites/stats/stats-post-detail/index.jsx]:
```suggestion
```

**Review by jsnmoon — APPROVED** [review]
Feel free to ship the PR after removing some extraneous code that could be pruned from the `StatsPostDetail` component.

**Comment by dognose24** [comment]
> removing some extraneous code that could be pruned from the StatsPostDetail component.

They were pruned in [6fefb6e](https://github.com/Automattic/wp-calypso/pull/71530/commits/6fefb6e0dda09483b25e12bfad30e6821bb2c444).

**Review by jsnmoon — APPROVED** [review]

**Comment by jeffgolenski** [comment]
@dognose24 Another wonderful addition. Thanks for this. The only issue I'm seeing right now is that the All-time stats box breaks the layout when the title is too long. 

Example: This post title in english occupies 3 lines, which pushes the card to break the balance with the card next to it

Screen Shot 2023-01-04 at 4 04 01 PM

Can we:

- [ ] Limit the displayed title to 48 characters max (not including spaces) and then insert an ellipsis "..." when it hits that max character count?

So it looks like this...
Screen Shot 2023-01-04 at 4 04 41 PM

### Mobile

I'm also noticing that we need a bit of work on the mobile views (tablet and phone). 

Screen Shot 2023-01-04 at 4 12 11 PM
Screen Shot 2023-01-04 at 4 11 53 PM


Can we:

- [ ] Make both of these cards full width at a higher tablet breakpoint? 
- [ ] Ensure the featured image is displayed at 84 x 84 pixels on mobile?

Screen Shot 2023-01-04 at 4 15 07 PM


cc @keoshi

**Comment by dognose24** [comment]
> @dognose24 Another wonderful addition. Thanks for this. The only issue I'm seeing right now is that the All-time stats box breaks the layout when the title is too long.

@jeffgolenski Thanks for your feedback! Sounds like an excellent plan for the mobile layout! I will move them to the task and tackle them in the follow-up PR.

**Pull request merged**